shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/rewrite/parameter/impl/ShardingGeneratedKeyInsertValueParameterRewriter.java

```diff
@@ -34,13 +34,15 @@
  */
 @Setter
 public final class ShardingGeneratedKeyInsertValueParameterRewriter implements ParameterRewriter<InsertStatementContext> {
-    
+
     @Override
     public boolean isNeedRewrite(final SQLStatementContext sqlStatementContext) {
-        return sqlStatementContext instanceof InsertStatementContext 
-                && ((InsertStatementContext) sqlStatementContext).getGeneratedKeyContext().isPresent() && ((InsertStatementContext) sqlStatementContext).getGeneratedKeyContext().get().isGenerated();
+        return sqlStatementContext instanceof InsertStatementContext
+                && ((InsertStatementContext) sqlStatementContext).getGeneratedKeyContext().isPresent()
+                && ((InsertStatementContext) sqlStatementContext).getGeneratedKeyContext().get().isGenerated()
+                && !((InsertStatementContext) sqlStatementContext).getGeneratedKeyContext().get().getGeneratedValues().isEmpty();
     }
-    
```

Comment:
@zjcnb Please keep indents consistent with the previous one.

Reply:
ok